### PR TITLE
Remove the unstable Float trait

### LINKED ISSUE #32110: Tracking issue for libcore prelude traits
These are currently all unstable but part of the libcore prelude.

This is currently intentional, but the fact that they need to be unstable is somewhat unsettling. Arguably all unstable features in libcore should be on the path to deprecation or stabilization at some point.

### PR DESCRIPTION
**Pull request by SimonSapin**
Following up to #49896 and #50629. Fixes #32110.

### CODE REVIEW

File: src/libcore/num/dec2flt/rawfp.rs [hunk 1]
```diff
@@ -56,22 +56,28 @@ impl Unpacked {
 /// Should **never ever** be implemented for other types or be used outside the dec2flt module.
 /// Inherits from `Float` because there is some overlap, but all the reused methods are trivial.
```
Comment (fbstj) [inline src/libcore/num/dec2flt/rawfp.rs]: same as before "inherits from `Float`" is no longer true

File: src/libcore/num/dec2flt/rawfp.rs [hunk 2]
```diff
@@ -56,22 +56,28 @@ impl Unpacked {
...
     + Div<Output=Self>
     + Neg<Output=Self>
-where
-    Self: Float<Bits = <Self as RawFloat>::RawBits>
 {
     const INFINITY: Self;
     const NAN: Self;
     const ZERO: Self;
 
     /// Same as `Float::Bits` with extra traits.
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply src/libcore/num/dec2flt/rawfp.rs]: Good point, fixed.

File: src/libcore/num/dec2flt/rawfp.rs (hunk 1 above)
Comment (SimonSapin) [reply src/libcore/num/dec2flt/rawfp.rs]: I’ve now removed this sentence.

**Comment by withoutboats** [comment]
I have no context

@bors r? @alexcrichton

**Pull request merged**